core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -881,17 +881,74 @@ object JsonAST {
      */
     val compactJs = RenderSettings(0, jsEscapeChars)
   }
+
+  /**
+   * Parent trait for double renderers, which decide how doubles contained in
+   * a JDouble are rendered to JSON string.
+   */
+  sealed trait DoubleRenderer extends Function1[Double,String] {
+    def apply(double: Double): String
+  }
+  /**
+   * A `DoubleRenderer` that renders special values `NaN`, `-Infinity`, and
+   * `Infinity` as-is using `toString`. This is not strictly valid JSON, but
+   * JavaScript can eval it. Other double values are also rendered the same
+   * way.
```

Comment:
It's the emphasis on Lift specifically not being able to reparse it. Once could read this and interpret it as "This isn't strictly valid JSON even though we can write/parse it" which would not be true.